coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -3939,10 +3939,18 @@ def dim(context, node):
 def min(context, node):
     inputs = _get_inputs(context, node)
 
-    if len(inputs) == 3:
+    # mimic functionality from https://pytorch.org/docs/stable/generated/torch.min.html
+    if len(inputs) == 1:
+        value = mb.reduce_min(x=inputs[0], axes=None, name=node.name)
+        context.add(value)
+    elif len(inputs) == 2 and inputs[0].shape == inputs[1].shape:
+        value = mb.minimum(x=inputs[0], y=inputs[1], name=node.name)
+        context.add(value)
+    else:
+        assert(len(inputs) <= 3)
```

Comment:
I think assert(len(inputs) == 3) makes more sense here.

Reply:
The reason for `<=3` is that we only want to do elementwise min/max if you receive two inputs that are both tensors and of the same shape: `torch.min(input, other, *, out=None) → Tensor`, however you could also receive `torch.min(input, dim, keepdim=False, *, out=None)` where dim is an int according to the [Pytorch docs](https://pytorch.org/docs/stable/generated/torch.min.html). 

The second branch in this conditional, checks for `len(inputs) == 2 and inputs[0].shape == inputs[1].shape`, otherwise it falls into this else case where it can still have either 2 or 3 inputs.

Not sure why these ops have multiple function signatures in Pytorch, but I tired to replicate it as accurately as possible. Let me know if you'd like me to re-write it though :)

Reply:
Ok, thanks for explaining that.